### DISCUSSION TITLE
burn and savings

### DISCUSSION THREAD

**Discussion by veebull** [Cyfrin/foundry-full-course-cu#664] [General]
Hello everybody!
How ethereum calculate **burnt gas** and part of **tx savings**. Please write a formula. Thanks

**Comment by Tanmay53** [comment]
Here's what I got from ChatGPT :

> Ethereum's gas fee system and the concept of "burning" gas is related to the Ethereum blockchain's transaction validation and consensus mechanism, which is based on Proof of Work (PoW). Gas is a unit of measurement used to quantify the computational work required to execute operations or smart contracts on the Ethereum network. The gas fee, denominated in Ether (ETH), is paid by users to compensate miners for including their transactions in a block and executing the associated computations.
> 
> When a transaction is executed on the Ethereum network, it consumes a certain amount of gas. Each operation in a smart contract consumes a specific amount of gas based on its complexity and computational requirements. This gas consumption is measured in "gas units." The total gas fee for a transaction is calculated by multiplying the gas used by the current gas price (in Gwei, a subunit of ETH).
> 
> 
> The formula for calculating the gas fee for a transaction is:
> 
> Gas Fee (in ETH) = Gas Used × Gas Price (in Gwei) / 1,000,000,000
> 
> Where:
> - Gas Used: The amount of gas units consumed by the transaction.
> - Gas Price: The price per gas unit in Gwei (1 Gwei = 10^-9 ETH).
> 
> Now, regarding the concept of "burning" gas, when a transaction is executed, the gas fee paid by the sender is not directly received by the miner. Instead, the gas fee is temporarily held in an account associated with the miner, and a portion of the gas fee is destroyed or "burned." This process is designed to reduce the overall supply of Ether over time, making it a deflationary mechanism.
> 
> The formula for calculating the amount of gas that is "burned" in a transaction is:
> 
> Burnt Gas (in ETH) = Gas Used × (Gas Price - Base Fee) / 1,000,000,000
> 
> 
> Where:
> - Base Fee: The minimum fee required to include a transaction in a block, which is dynamically adjusted by the network to maintain a stable block size.
> 
> It's important to note that Ethereum has been undergoing transitions, and Ethereum 2.0 (Eth2) aims to transition from Proof of Work to Proof of Stake (PoS), which changes the consensus mechanism and how gas fees are calculated. Eth2 uses a different mechanism to reward validators and process transactions, which could impact the details of gas calculation and fee distribution. Always refer to the most current Ethereum documentation and resources for the latest information on gas calculation and network updates.
